qemu/tests/qemu_guest_agent.py

```diff
@@ -885,6 +895,42 @@ def gagent_check_fsfreeze(self, test, params, env):
         # Finally, do something after thaw.
         self._action_after_fsthaw(test, params, env)
 
+    @error_context.context_aware
+    def gagent_check_hotplug_virtio_console(self, test, params, env):
+        """
+        hotplug guest agent device
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+
+        char_backend = params["char_backend"]
+        char_id = params["char_id"]
+        char_path = params["char_path"]
+        dev_driver = params["dev_driver"]
+        dev_id = params["dev_id"]
+        dev_name = params["dev_name"]
+        self.vm.verify_alive()
+        error_context.context("hotplug guest agent device", logging.info)
+        chardev = qdevices.CharDevice(params=params)
+        chardev.set_param("backend", char_backend)
+        chardev.set_param("id", char_id)
+        chardev.set_param("path", char_path)
```

Comment:
You have passed params to `chardev`, why not define above parameters directly in params?

Reply:
Because "id" has already existed in previous params. 
id = smp_32.4096m.repeat1.run_test.Host_RHEL.m7.u5.raw.vi rtio_blk.up.virtio_net.RHEL.7.5.x86_64.io-github-autotest-qemu.qemu_guest_agent.default.check_hotplug_virtio_console

It is needed to be set again.

Reply:
You may define id for your chardev like `id_${chardev_name}` to specify all specific parameters for chardev and then pass the param as `params.object_params(${chardev_name})`

Reply:
Yes, it is a better way.